legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/common/DatasetsAbstract.java

```diff
@@ -43,4 +43,6 @@
     Optional<Dataset> tempDataset();
 
     Optional<Dataset> tempDatasetWithDeleteIndicator();
+
+    Optional<Dataset> stagingDatasetWithoutDuplicates();
```

Comment:
Wondering if we need this as a dataset? Can't we always use the same structure as staging table as use our own naming convention for the table?

Reply:
Yes it is going to be always the same structure as the staging table. It is done this way to be consistent with our approach with all the other temp tables (they have the same structure as the main table too). Our reason for exposing an interface for all these temp tables was so that AS could have control of them (they wanted to create them and keep them). At the same time they are also not mandatory, if user doesnt provide them, we will create default tables for all these temp tables with our own naming convention.